h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_7798_overlapped_knots.py

```diff
@@ -0,0 +1,36 @@
+import h2o
+import numpy as np
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+from tests import pyunit_utils
+
+def knots_error():
+    # load and prepare California housing dataset
+    np.random.seed(1234)
+    data = h2o.H2OFrame(
+        python_obj={'C1': list(np.random.randint(0, 9, size=1000)),
+                    'target': list(np.random.randint(0, 2, size=1000))
+                    })
+    # use only 3 features and transform into classification problem
+    feature_names = ['C1']
+    data['target'] = data['target'].asfactor()
+    # split into train and validation sets
+    train, test = data.split_frame([0.8], seed=1234)
+    # build the GAM model
+    h2o_model = H2OGeneralizedAdditiveEstimator(family='binomial',
+                                                gam_columns=feature_names,
+                                                scale=[1],
+                                                num_knots=[10],
+                                                )
+    try:
+        h2o_model.train(x=feature_names, y='target', training_frame=train)
+        assert False, "Number of knots validation should have failed"
+    except:
+        print("Error correctly raised when cardinality < num_knots")
+    else:
```

Comment:
Karthik:  Please check the actual error message and make sure you got the correct one.  You can do something like this:

    try:
        glm.makeGLMModel(model=m,coefs=coefs)
        assert False, "Should have throw exception of bad coefficient length"
    except Exception as ex:
        print(ex)
        temp = str(ex)
        assert ("Server error java.lang.IllegalArgumentException:" in temp) and \
               ("model coefficient length 9 is different from coefficient provided by user ") in temp, \
            "Wrong exception was received."
        print("coefficient test passed!")